src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -543,145 +545,131 @@ private async ValueTask FinishBatchOfDocumentsAsync()
+                }
             }
 
-            private class UpdateClusterTransactionIndex : TransactionOperationsMerger.MergedTransactionCommand, TransactionOperationsMerger.IReplayableCommandDto<UpdateClusterTransactionIndex>
+            private async ValueTask SendAddOrUpdateCommandsAsync(JsonOperationContext context)
             {
-                public long Index;
-                public readonly List<string> DocumentIds = new List<string>();
-                public readonly List<string> CompareExchangeToRemove = new List<string>();
-
-                protected override long ExecuteCmd(DocumentsOperationContext context)
-                {
-                    long count = 0;
-                    foreach (var docId in DocumentIds)
-                    {
-                        using var document = context.DocumentDatabase.DocumentsStorage.Get(context, docId);
-                        if (document == null)
-                        {
-                            CompareExchangeToRemove.Add(docId);
-                            continue;
-                        }
-
-                        count++;
-                       
-                        var changeVectorList = document.ChangeVector.ToChangeVectorList();
-                        for (int i = 0; i < changeVectorList.Count; i++)
-                        {
-                            if (changeVectorList[i].DbId != context.DocumentDatabase.ClusterTransactionId) 
-                                continue;
-                            
-                            changeVectorList.RemoveAt(i);
-                            i--;
-                        }
-                        changeVectorList.Add(new ChangeVectorEntry
-                        {
-                            Etag = Index,
-                            DbId = context.DocumentDatabase.ClusterTransactionId,
-                            NodeTag = ChangeVectorParser.TrxnInt
-                        });
-                        var newChangeVector = changeVectorList.SerializeVector();
-
-                        var newDoc = context.ReadObject(document.Data, docId, BlittableJsonDocumentBuilder.UsageMode.ToDisk);
-                        context.DocumentDatabase.DocumentsStorage.Put(context, docId, null, newDoc, document.LastModified.Ticks,
-                            changeVector: newChangeVector,
-                            flags: document.Flags);
-                    }
-
-                    return count;
-                }
+                if (_compareExchangeAddOrUpdateCommands.Count == 0)
+                    return;
 
-                public override TransactionOperationsMerger.IReplayableCommandDto<TransactionOperationsMerger.MergedTransactionCommand> ToDto(JsonOperationContext context)
+                var (aoucecIndex, _) = await _database.ServerStore.SendToLeaderAsync(new AddOrUpdateCompareExchangeBatchCommand(_compareExchangeAddOrUpdateCommands, context, RaftIdGenerator.DontCareId));
```

Comment:
`aoucec` ?

Reply:
**A**dd**O**r**U**pdate**C**ompare**E**xchange**C**ommand
a___o__u______c_________e________c

Reply:
let's call it simply `index` or do `var result = ... `